modelon/impact/client/entities/workspace.py

```diff
@@ -17,13 +20,63 @@
 from modelon.impact.client.entities.model import Model
 from modelon.impact.client.entities.model_executable import ModelExecutable
 from modelon.impact.client.entities.experiment import Experiment
+from modelon.impact.client.entities.project import Project, ProjectDefinition
 
 
 logger = logging.getLogger(__name__)
 
 ExperimentDefinition = Union[
     SimpleModelicaExperimentDefinition, SimpleFMUExperimentDefinition, Dict[str, Any],
 ]
+DEFAULT_PROJECT_NAME = 'NewProject'
+
+
+@dataclass
+class ProjectEntry:
+    id: str
+    disabled: bool
+    disabled_content: List[str]
+
+    @classmethod
+    def from_dict(cls, data):
+        reference = data.get('reference')
+        return cls(
+            id=reference.get('id'),
+            disabled=data.get('disabled'),
+            disabled_content=data.get('disabledContent'),
+        )
+
+
+@dataclass
+class WorkspaceDefinition:
+    name: str
+    format: str
+    description: str
+    created_by: str
+    created_at: int
+
+    default_project_id: str
+
+    projects: List[ProjectEntry]
+
+    dependencies: List[ProjectEntry]
+
+    @classmethod
+    def from_dict(cls, data):
+        projects = data.get('projects', [])
+        dependencies = data.get('dependencies', [])
```

Comment:
I think it would be good to let the workspace definition keep the `data` as a private attribute (probably easier to do this without having it as a data class) so that the data is just passed on when sharing or in other way then serializing and passing on the data.